src/apps/search-header/search-header.tsx

```diff
@@ -1,27 +1,156 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
+import { useCombobox, UseComboboxStateChange } from "downshift";
+import {
+  SuggestionsFromQueryStringQuery,
+  useSuggestionsFromQueryStringQuery
+} from "../../core/dbc-gateway/generated/graphql";
 import SearchBar from "../../components/search-bar/search-bar";
+import { Autosuggest } from "../../components/autosuggest/autosuggest";
+import { Suggestion } from "../../components/autosuggest-text/autossugest-text";
 
 export interface SearchHeaderProps {
   searchHeaderUrl?: string;
   altText?: string;
-  inputPlaceholder?: string;
+  inputPlaceholderText?: string;
+  stringSuggestionAuthorText?: string;
+  stringSuggestionWorkText?: string;
+  stringSuggestionTopicText?: string;
 }
 
 const SearchHeader: React.FC<SearchHeaderProps> = ({
   searchHeaderUrl = "/search",
   altText = "search icon",
-  inputPlaceholder = "Search here"
+  inputPlaceholderText = "Search here",
+  stringSuggestionAuthorText = "Author",
+  stringSuggestionWorkText = "Work",
+  stringSuggestionTopicText = "Topic"
 }) => {
-  const [q, setQ] = useState("");
+  const [q, setQ] = useState<string>("");
+  const [suggestItems, setsuggestItems] = useState<any[]>([]);
+  const [currentlySelectedItem, setCurrentlySelectedItem] = useState<any>("");
+  const [isAutosuggestOpen, setIsAutosuggestOpen] = useState<boolean>(false);
+  const {
+    data,
+    isLoading,
+    status
+  }: {
+    data: SuggestionsFromQueryStringQuery | undefined;
+    isLoading: boolean;
+    status: string;
+  } = useSuggestionsFromQueryStringQuery({ q });
+
+  // once the query returns data, we set it into our useSate
+  useEffect(() => {
+    if (data) {
+      const arayOfResults = data.suggest.result;
+      setsuggestItems(arayOfResults);
+    }
+  }, [data]);
+
+  // if there are at least 3 chars in the search-field, we open the autosuggest
+  useEffect(() => {
+    if (q) {
+      if (q.length > 2) {
+        setIsAutosuggestOpen(true);
+      }
+    } else {
+      setIsAutosuggestOpen(false);
+    }
+  }, [q]);
+
+  function handleSelectedItemChange(
+    changes: UseComboboxStateChange<Suggestion>
+  ) {
+    if (!changes.selectedItem) {
+      return;
+    }
+    switch (changes.selectedItem.__typename) {
+      case "Creator":
+        setCurrentlySelectedItem(changes.selectedItem.name);
+        break;
+      case "Subject":
+        setCurrentlySelectedItem(changes.selectedItem.value);
+        break;
+      default:
+        setCurrentlySelectedItem(changes.selectedItem.title);
+    }
```

Comment:
Consider adding a function for getting a string for a suggestion. This seems to be a repeating pattern.

Reply:
Great suggestion! Done :)